### PR TITLE
add duckdb support

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/holoviz/hvplot/issues/1397

```python
import hvplot.duckdb

import pandas as pd
import duckdb

df = pd.DataFrame({
    'x': [1, 2, 3],
    'y': [4, 5, 6]
})

df.to_parquet('test.parquet')

connection = duckdb.connect(database=':memory:', read_only=False)
connection.from_parquet("test.parquet").hvplot(
    x='x', y='y', kind='scatter'
)
```

image

### CODE REVIEW

**Comment by MarcSkovMadsen** [comment]
Is the long term goal to implement a real duckdb backend for more efficient usage?

**Comment by ahuang11** [comment]
Can you elaborate on `real duckdb backend`?

This PR doesn't change the converter at all.

**Comment by MarcSkovMadsen** [comment]
Yes.

Is the plan to implement a real HoloViz backend such that you only query the data needed for the plot? For example when using `groupby`.

**Comment by ahuang11** [comment]
I'm not sure I follow; this PR allows hvplot off of a DuckDBRelation, which is already the result of a query.

connection > relation (select * from table where ...) > hvplot

**Comment by ahuang11** [comment]
Was slightly mistaken, but a little modification works:
https://github.com/holoviz/hvplot/pull/1398/files#diff-a47979cba5da76fbc1aec07d45b4403241da5cfbbbf6b36652f75c52fb644824R360

File: hvplot/duckdb.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    from . import hvPlotTabular, post_patch
+
+    try:
+        import duckdb
+    except ImportError:
+        raise ImportError(
+            'Could not patch plotting API onto DuckDB. DuckDB could not be imported.'
+        )
+
+    # Patching for DuckDBPyRelation
```
Comment (ahuang11) [inline hvplot/duckdb.py:14]:
```suggestion
    # Patching for DuckDBPyRelation and DuckDBConnection
```

File: hvplot/duckdb.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+"""Adds the `.hvplot` method to duckdb.DuckDBPyRelation"""
```
Comment (ahuang11) [inline hvplot/duckdb.py:1]:
```suggestion
"""Adds the `.hvplot` method to duckdb.DuckDBPyRelation and duckdb.DuckDBPyConnection"""
```

File: hvplot/duckdb.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    from . import hvPlotTabular, post_patch
+
+    try:
+        import duckdb
+    except ImportError:
+        raise ImportError(
+            'Could not patch plotting API onto DuckDB. DuckDB could not be imported.'
+        )
+
+    # Patching for DuckDBPyRelation and DuckDBConnection
```
Comment (ahuang11) [inline hvplot/duckdb.py:14]:
```suggestion
    # Patching for DuckDBPyRelation and DuckDBPyConnection
```

**Comment by ahuang11** [comment]
Oh I guess it is converting to pandas object first
https://github.com/holoviz/hvplot/pull/1398/files#diff-be8c4d86a0e1601a53aca277191c3a5d8e160fc39c2b4d75396c83b6a38ae610R15

File: hvplot/plotting/__init__.py [hunk 4]
```diff
@@ -11,6 +11,7 @@
 
 @with_hv_extension
 def plot(data, kind, **kwargs):
+    print(data)
```
Comment (ahuang11) [inline hvplot/plotting/__init__.py:14]:
```suggestion
```

File: doc/index.md [hunk 5]
```diff
@@ -116,6 +117,23 @@ align: center
 ---
 :::
 
+:::{tab-item} DuckDB
```
Comment (maximlt) [inline doc/index.md:120]: It doesn't seem like DuckDB is used in this example?

File: doc/user_guide/Integrations.ipynb [hunk 6]
```diff
@@ -327,6 +327,113 @@
...
+    ":::"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0d6460d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
```
Comment (maximlt) [inline doc/user_guide/Integrations.ipynb:354]: How about merging these two code cells? It'd look a little cleaner in the docs.

File: doc/user_guide/Integrations.ipynb [hunk 7]
```diff
@@ -327,6 +327,113 @@
...
+    "import numpy as np\n",
+    "import hvplot.pandas  # noqa\n",
```
Comment (maximlt) [inline doc/user_guide/Integrations.ipynb:356]: No need to import `hvplot.pandas`.

**Review by maximlt — COMMENTED** [review]
I'm ok on adding DuckDB support. It looks like you should update the diagram on the landing page too, unless you consider this experimental first.
image

**Comment by maximlt** [comment]
@ahuang11 can you also update this PR based on https://github.com/holoviz/hvplot/pull/1359 now it has been merged?

File: doc/index.md (hunk 5 above)
Comment (ahuang11) [reply doc/index.md:120]: You're right :D I copied and pasted but forgot to add the duckdb.from_df part; added now.

**Comment by ahuang11** [comment]
Okay I think this is ready. I added the diagrams + added changes from #1359

File: doc/assets/diagram.svg
Comment (maximlt) [inline doc/assets/diagram.svg:1]: Oh have you just replaced `Intake` by `DuckDB`? Have you recently removed `Intake` support?

File: doc/index.md [hunk 8]
```diff
@@ -116,6 +117,24 @@ align: center
...
+```python
+import duckdb
+import hvplot.duckdb
+from bokeh.sampledata.autompg import autompg_clean as df
+
+df_duckdb = duckdb.from_df(df)
+table = df_duckdb.groupby(['origin', 'mfr'])['mpg'].mean().sort_values().tail(5)
+table.hvplot.barh('mfr', 'mpg', by='origin', stacked=True)
+```
+```{image} ./_static/home/dask.gif
```
Comment (maximlt) [inline doc/index.md:130]: `dask.gif` shows `bokeh.sampledata.penguins`.

```suggestion
```{image} ./_static/home/pandas.gif
```

File: doc/assets/diagram.svg
Comment (hoxbro) [reply doc/assets/diagram.svg:1]: I think he has moved intake up. 

Some notes: 
- Arrow are missing
- Matplotlib is not straight. 

image

Comment (maximlt) [reply doc/assets/diagram.svg:1]: Gosh I'm blind!

**Review by philippjfr — APPROVED** [review]

File: hvplot/plotting/__init__.py (hunk 4 above)
Comment (hoxbro) [inline hvplot/plotting/__init__.py:14]:
```suggestion
```

You likely force pushed

File: hvplot/tests/testplotting.py [hunk 9]
```diff
@@ -50,3 +54,28 @@ def test_pandas_dataframe_plot_does_not_implement_pie(self):
 class TestPandasHvplotPlotting(TestPandasHoloviewsPlotting):
     def setUp(self):
         pd.options.plotting.backend = 'hvplot'
+
+
+def test_plot_supports_polars():
```
Comment (hoxbro) [inline hvplot/tests/testplotting.py:59]: Why did you add this? Maybe you didn't know about 

https://github.com/holoviz/hvplot/blob/266565b9b54414b7eed62203dcfcdc460f098505/hvplot/tests/plotting/testcore.py#L84

Comment (ahuang11) [reply hvplot/tests/testplotting.py]: I guess I was confused by the merge conflict.

**Review by maximlt — APPROVED** [review]
